client/finality-grandpa-warp-sync/src/proof.rs

```diff
@@ -42,11 +43,21 @@ pub struct AuthoritySetChangeProof<Block: BlockT> {
 	pub justification: GrandpaJustification<Block>,
 }
 
+/// Represents the current state of the warp sync, namely whether it is considered
+/// finished, i.e. we have proved everything up until the latest authority set, or not.
+/// When the warp sync is finished we might optionally provide a justification for the
+/// latest finalized block, which should be checked against the latest authority set.
+#[derive(Debug, Decode, Encode)]
+pub enum WarpSyncFinished<Block: BlockT> {
```

Comment:
Let me know if this is too annoying to deal with but we really need the tree states: finished w/ best justification, finished without best justification (since the latest authority set might not have finalized anything yet), unfinished.

Reply:
Could you maybe invert the enum variants? Right now `Yes` is encoded with 0 and `No` with 1, which is weird